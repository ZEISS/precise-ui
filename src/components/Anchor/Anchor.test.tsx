import * as React from 'react';
import * as enzyme from 'enzyme';
import { Anchor } from './';

describe('<Anchor />', () => {
  it('should render an empty <Anchor> component', () => {
    const wrapper = enzyme.shallow(<Anchor />);
    expect(wrapper.text()).toBe('');
  });

  it('should render an <Anchor> component with href', () => {
    const wrapper = enzyme.shallow(<Anchor href="foo">Text</Anchor>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render an <Anchor> component with to', () => {
    const wrapper = enzyme.shallow(<Anchor to="foo">Text</Anchor>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render an <Anchor> component with to and href', () => {
    const wrapper = enzyme.shallow(
      <Anchor to="foo" href="bar">
        Text
      </Anchor>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render an <Anchor> component with onClick', () => {
    const wrapper = enzyme.shallow(<Anchor onClick={() => console.log()}>Text</Anchor>);
    expect(wrapper).toMatchSnapshot();
  });
});
