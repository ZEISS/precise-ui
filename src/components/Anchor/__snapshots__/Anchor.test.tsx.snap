// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Anchor /> should render an <Anchor> component with href 1`] = `
<a
  href="foo"
  onClick={[Function]}
>
  Text
</a>
`;

exports[`<Anchor /> should render an <Anchor> component with onClick 1`] = `
<a
  href="#"
  onClick={[Function]}
>
  Text
</a>
`;

exports[`<Anchor /> should render an <Anchor> component with to 1`] = `
<a
  href="foo"
  onClick={[Function]}
>
  Text
</a>
`;

exports[`<Anchor /> should render an <Anchor> component with to and href 1`] = `
<a
  href="bar"
  onClick={[Function]}
>
  Text
</a>
`;
